hnn_core/viz.py

```diff
@@ -1255,19 +1264,40 @@ def plot_laminar_csd(times, data, contact_labels, ax=None, colorbar=True,
         The matplotlib figure handle.
     """
     import matplotlib.pyplot as plt
+    from scipy.interpolate import RectBivariateSpline
+
     if ax is None:
         _, ax = plt.subplots(1, 1, constrained_layout=True)
 
-    im = ax.pcolormesh(times, contact_labels, np.array(data),
-                       cmap="jet_r", shading='auto')
-    ax.set_title("CSD")
+    if sink[0] == 'b':
+        cmap = "jet"
+    elif sink[0] == 'r':
+        cmap = "jet_r"
+
+    if interpolation == 'spline':
+        # create interpolation function
+        interp_data = RectBivariateSpline(times, contact_labels, data.T)
+        # increase number of contacts
+        new_depths = np.linspace(contact_labels[0], contact_labels[-1],
+                                 contact_labels[-1] - contact_labels[0])
+        # interpolate
+        data = interp_data(times, new_depths).T
+    elif not interpolation:
+        data = data
+        new_depths = contact_labels
+
+    # if vmin and vmax are both None, set colormap such that green = zero
+    if vmin is None and vmax is None:
+        vmin = -np.max(np.abs(data))
+        vmax = np.max(np.abs(data))
+
+    im = ax.pcolormesh(times, new_depths, data,
+                       cmap=cmap, shading='auto', vmin=vmin, vmax=vmax)
```

Comment:
wondering if we should use `imshow`? I know this is how it was implemented originally. The advantage of `imshow` is that you can mouseover to see the value.

Reply:
I tried this and imshow gets messy when interpolation = None. The aspect ratio of the axes has to be set, the origin has to be flipped, and the ticklabels for x and y have to be set. Maybe more complicated that necessary for this PR?

Reply:
oh dear! agreed